approval_polls/views.py

```diff
@@ -30,12 +33,19 @@ def getPolls(request, poll_list, render_page):
       polls = paginator.page(paginator.num_pages)
   return render(request, render_page, {'latest_poll_list' : polls})
 
-class DetailView(generic.DetailView):
+class DetailView(generic.DeleteView):
   model = Poll
   template_name = 'approval_polls/detail.html'
   def get_queryset(self):
       return Poll.objects.filter(pub_date__lte=timezone.now())
 
+  @staticmethod
```

Comment:
@Brianjr0428 That's a good question. I think reading about [REST](https://en.wikipedia.org/wiki/Representational_state_transfer) will be helpful at this stage, especially the section on [Uniform Interface](https://en.wikipedia.org/wiki/Representational_state_transfer#Uniform_interface).

The basic gist is that resources in a web application are identified and manipulated using a single URI and thus a single view and a set of HTTP methods (GET, PUT, POST, DELETE).

In our case, the resource we want to identify and subsequently manipulate is a **poll**. The URI for each poll is of the form `/approval_polls/poll_id`. Now the actions we can take on a poll are

1. Retrieve a poll to view it (`GET /approval_polls/poll_id`)
2. Update properties of a poll like publish date, title, choices etc. (`PUT /approval_polls/poll_id`)
3. Delete a poll (`DELETE /approval_polls/poll_id`)

New polls are created using the URI `/approval_polls/create` with a POST request as we don't know the poll_id before creating the poll. Using `GET /approval_polls/create` will simply retrieve an empty form that you can fill in to create a new poll.

As part of this PR, you are implementing 1 and 3. This means we need a single class-based view for the resource **poll** with two http methods servicing that view (GET and DELETE). To enable the edit option, you will have to add another method for the PUT request.

Another important point is that the views are not tied to the *pages* but to *resources*.

If you're still finding this confusing, then I suggest you take some time off to read more about the wonderful world of HTTP and how web applications are designed around it using REST. The course on [Web Development on Udacity](https://www.udacity.com/course/web-development--cs253) helped me a lot in this regard.